webhooks-extension/endpoints/sink.go

```diff
@@ -0,0 +1,342 @@
+	pipelineResource := v1alpha1.PipelineResource{
+		ObjectMeta: metav1.ObjectMeta{Name: name, Namespace: namespace},
+		Spec: v1alpha1.PipelineResourceSpec{
+			Type:   resourceType,
+			Params: params,
+		},
+	}
+	resourcePointer := &pipelineResource
+	return resourcePointer
+}
+
+/* Create a new PipelineRun - repoUrl, resourceBinding and params can be nill depending on the Pipeline
+each PipelineRun has a 1 hour timeout: */
+func definePipelineRun(pipelineRunName, namespace, saName, repoURL string,
+	pipeline v1alpha1.Pipeline,
+	triggerType v1alpha1.PipelineTriggerType,
+	resourceBinding []v1alpha1.PipelineResourceBinding,
+	params []v1alpha1.Param) (*v1alpha1.PipelineRun, error) {
+
+	gitServer, gitOrg, gitRepo := "", "", ""
+	err := errors.New("")
+	if repoURL != "" {
+		gitServer, gitOrg, gitRepo, err = getGitValues(repoURL)
+		if err != nil {
+			log.Printf("there was an error getting the Git values: %s", err)
+			return &v1alpha1.PipelineRun{}, err
+		}
+	}
+
+	pipelineRunData := v1alpha1.PipelineRun{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      pipelineRunName,
+			Namespace: namespace,
+			Labels: map[string]string{
+				"app":          "devops-knative",
+				gitServerLabel: gitServer,
+				gitOrgLabel:    gitOrg,
+				gitRepoLabel:   gitRepo,
+			},
+		},
+
+		Spec: v1alpha1.PipelineRunSpec{
+			PipelineRef: v1alpha1.PipelineRef{Name: pipeline.Name},
+			// E.g. v1alpha1.PipelineTriggerTypeManual
+			Trigger:        v1alpha1.PipelineTrigger{Type: triggerType},
+			ServiceAccount: saName,
+			Timeout:        &metav1.Duration{Duration: 1 * time.Hour},
+			Resources:      resourceBinding,
+			Params:         params,
+		},
+	}
+	pipelineRunPointer := &pipelineRunData
+	return pipelineRunPointer, nil
+}
+
+// Returns the git server excluding transport, org and repo
+func getGitValues(url string) (gitServer, gitOrg, gitRepo string, err error) {
+	repoURL := ""
+	if url != "" {
+		url = strings.ToLower(url)
+		if strings.Contains(url, "https://") {
+			repoURL = strings.TrimPrefix(url, "https://")
+		} else {
+			repoURL = strings.TrimPrefix(url, "http://")
+		}
+	}
+
+	// example at this point: github.com/tektoncd/pipeline
+	numSlashes := strings.Count(repoURL, "/")
+	if numSlashes < 2 {
+		return "", "", "", errors.New("Url didn't match the requirements (at least two slashes)")
+	}
+	repoURL = strings.TrimSuffix(repoURL, "/")
+
+	gitServer = repoURL[0:strings.Index(repoURL, "/")]
+	gitOrg = repoURL[strings.Index(repoURL, "/")+1 : strings.LastIndex(repoURL, "/")]
+	gitRepo = repoURL[strings.LastIndex(repoURL, "/")+1:]
+
+	return gitServer, gitOrg, gitRepo, nil
+}
+
+func getDateTimeAsString() string {
+	return strconv.FormatInt(time.Now().Unix(), 10)
+}
+
+// SinkWebService returns the liveness web service
+func SinkWebService(r Resource) *restful.WebService {
+	ws := new(restful.WebService)
+	ws.
+		Path("/")
+	ws.Route(ws.POST("").To(r.handleWebhook))
+
+	return ws
+}
```

Comment:
General comment that might be easy to sort now is capitalisation of log.Print and log.Printf messages ... there seems to be a lot of variations in the message starting with a capital.  It seems like a quick and easy thing to do here.

I'm not going to be critical of the code at this point as it is multiple authors and the work you were doing was for re-packaging

Reply:
Thanks, I think I got all of the logs capitalized 😁